src/webapi/rendering_context.rs

```diff
@@ -16,11 +18,187 @@ pub trait RenderingContext {
+/// identifying which of the twelve compositing operations to use.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/globalCompositeOperation)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum CompositeOperation {
+    SourceOver,
+    SourceIn,
+    SourceOut,
+    SourceAtop,
+    DestinationOver,
+    DestinationIn,
+    DestinationAtop,
+    Lighter,
+    Copy,
+    Xor,
+    Multiply,
+    Screen,
+    Overlay,
+    Darken,
+    Lighten,
+    ColorDodge,
+    ColorBurn,
+    HardLight,
+    SoftLight,
+    Difference,
+    Exclusion,
+    Hue,
+    Saturation,
+    Color,
+    Luminosity
+}
+
+/// The algorithm by which to determine if a point is inside a path or outside a path.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fill)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum FillRule {
+    NonZero,
+    EvenOdd
+}
+
+/// h=How the end points of every line are drawn.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineCap)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum LineCap {
+    Butt,
+    Round,
+    Square
+}
+
+/// determines how two connecting segments (of lines, arcs or curves) with non-zero lengths in a shape are 
+/// joined together (degenerate segments with zero lengths, whose specified endpoints and control points are 
+/// exactly at the same position, are skipped).
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/lineJoin)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum LineJoin {
+    Bevel,
+    Round,
+    Miter
+}
+
+/// A DOMString indicating how to repeat the image.
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/createPattern)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Repetition {
+    Repeat,
+    RepeatX,
+    RepeatY,
+    NoRepeat
+}
+
+/// Specifies text alignment
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textAlign)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TextAlign {
+    Left,
+    Right,
+    Center,
+    Start,
+    End
+}
+
+/// Text baseline being used when drawing text
+/// 
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/textBaseline)
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TextBaseline {
+    Top,
+    Hanging,
+    Middle,
+    Alphabetic,
+    Ideographic,
+    Bottom
+}
+
+error_enum_boilerplate! {
+    IndexSizeError,
+    SyntaxError
+}
+
+error_enum_boilerplate! {
+    InvalidStateError,
+    SyntaxError
+}
+
+error_enum_boilerplate! {
+    NotSupportedError,
+    SyntaxError
+}
+
+error_enum_boilerplate! {
+    TypeMismatchError,
+    SyntaxError
+}
+
+error_enum_boilerplate! {
+    NsErrorNotAvailable,
+    SecurityError
+}
```

Comment:
1. This doesn't seem to be used currently; do you intend to use it somewhere?
2. Isn't this Firefox-specific? (Due to the `Ns` prefix?)

Reply:
Yup! You're right, I missed that